roofit/roofitcore/inc/SpanCast.h

```diff
@@ -0,0 +1,111 @@
+// Author: Jonas Rembser, CERN, Jul 2021
```

Comment:
I don't know what the rules are here, it seems that everybody does what he or she wants in new RooFit files. So far, I have not added some header to new files and @lmoneta didn't complain about out :)

Reply:
There's no "rules" as such, but all RooFit files that are not tests do have a header ...

Reply:
... excluding a few external contributions maybe.

Reply:
Okay fine I don't mind putting a header, but which one? The old RooFit header or the general ROOT header? I see many files with some special RooFit header: https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooAbsBinning.h.

But I also see files that you added with the ROOT header: https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooWorkspaceHandle.h.

Should I put the ROOT header then?

Reply:
> Okay fine I don't mind putting a header, but which one? The old RooFit header or the general ROOT header? I see many files with some special RooFit header: https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooAbsBinning.h.

That should be most files, actually.

> 
> But I also see files that you added with the ROOT header: https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooWorkspaceHandle.h.
That's my 2nd month. Maybe I hadn't decided by then ... 😅 

> Should I put the ROOT header then?
Given the above: No?

Reply:
> I just remembered that `TRangeDynCast` and `TRangeStaticCast` are already a thing in ROOT, meant for use with `TCollection`s. I looked them up and I think they are generic enough to be used with RooFit collections too (but they suffer from similar non-STL-compliance issues as the first version of this PR). Maybe generalize/improve those rather than adding a RooFit-specific version? So everybody benefits.

I actually pointed specifically to those when bringing this up, so 👍. And yes, they need some love. And they also should be documented much better, see #8421.
I think it's fair to adapt them properly for use in RooFit, because then they will be used more actively, and get the love they deserve.